### ISSUE TITLE
[Core, Database, Globus] Implement Consent Validation and Storing in Core Service 

### ISSUE THREAD

**Issue by AronPerez** [ORNL/DataFed#1144]
## Description

The goal of this ticket is to:

I. Validating if a consent can be used on `y collection for x user`. This is done by:

1. Knowing what scopes/collection token(s) are necessary for consent. 
`urn:globus:auth:scope:transfer.api.globus.org:all` [[ref](https://docs.globus.org/api/transfer/overview/#data_access_consent)]
2. Checking if  **consent** exists for `x user on y collection` [[ref](https://globus-sdk-python.readthedocs.io/en/latest/authorization/scopes_and_consents/consents.html#reference)] in Arango.
  A. If consent exists, proceed to step 4. 
  B. If the consent DNE then query `v2/api/identities/{{identity}}/consents` (see example below)
4. Check consents expiration:
    A. If the user needs new consent, send URI. (follow https://github.com/ORNL/DataFed/issues/1116)
    B. Otherwise continue with existing consent and update in Arango as needed
5. Store consent as needed for `x user on y collection`.

## Example
[Doc](https://confluence.ccs.ornl.gov/display/DATAFED/GENERAL+-+Brainstorming+fix+to+Globus+Mapped+Collection+Consent?focusedCommentId=412555383#comment-412555383)

            GET [https://auth.globus.org/v2/api/identities/{{identity}}/consents](https://auth.globus.org/v2/api/identities/%7B%7Bidentity%7D%7D/consents)
            {{identity}} = identity_id
            Auth: Bearer <access_token> w/ transfer:all scopes
```JSON
            {
            "consents": [
                {
                "id": 13960386,
                "client": "71c0c547-02e6-44b4-9232-d7501147cdde",
                "status": "approved",
                "scope_name": "openid",
                "auto_approved": false,
                "created": "2024-11-06T19:21:56.787697+00:00",
                "atomically_revocable": false,
                "dependency_path": [
                    13960386
                ],
                "scope": "245cd2d7-4831-4ee4-a81d-f8892625e865",
                "allows_refresh": true,
                "updated": "2024-11-06T19:21:56.787697+00:00",
                "last_used": "2024-11-15T17:50:49.815903+00:00",
                "effective_identity": "0e4856cf-a652-42f1-8541-f2b78a70e607"
            },
```

## Resources
* [Python Globus SDK Consent Tree](https://globus-sdk-python.readthedocs.io/en/latest/authorization/scopes_and_consents/consents.html#globus_sdk.scopes.consents.ConsentForest)
* [Globus API Docs data access consent](https://docs.globus.org/api/transfer/overview/#data_access_consent)

**Comment by JoshuaSBrown** [comment]
You probably already incude this on checking consents. But to be explicit making sure that refresh tokens are valid. For instance after 6 months of inactivity, I'm pretty sure refresh tokens become invalid regardless of what consents were valid on them.

**Comment by JoshuaSBrown** [comment]
Oh, I see this is just the consent item.

**Comment by JoshuaSBrown** [comment]
So how is consent different from the tokens, are you thinking that a consent requirement could expire before a token becomes invalid?

**Comment by AronPerez** [comment]
https://github.com/ORNL/DataFed/issues/1120 exists

**Comment by JoshuaSBrown** [comment]
Another question, is there ever a case where you might grant different consents on the same collection?

**Comment by AronPerez** [comment]
> So how is consent different from the tokens, are you thinking that a consent requirement could expire before a token becomes invalid?

No. A “Consent” is a record, in Globus Auth, that an application is allowed to take some action on behalf of a user. That being said I thought it might be useful to look at Arango to see if we had consent.

Maybe as long as we:
1. Store the `access_token`/`refresh_token`, 
2. Try `ls` or `stat`, and r
3. Respond to the results

We won't need to save consent.

https://github.com/globus/globus-sdk-python/blob/main/docs/examples/minimal_transfer_script/transfer_consent_required_proactive.py#L48
